src/plugin.tsx

```diff
@@ -104,11 +107,29 @@ export class KalturaLivePlugin implements OnMediaUnload, OnRegisterUI, OnMediaLo
             this._contribServices.overlayManager.remove(this._overlayItem);
             this._overlayItem = null;
         }
+        if (this._videoEnded) {
+            this._reloadVideo();
+        }
+    };
+
+    private _handleOnEnd = () => {
+        this._videoEnded = true;
+        this._setOffline();
+    };
+
+    private _reloadVideo = () => {
+        this._videoEnded = false;
+        this._player._detachMediaSource();
+        this._player._attachMediaSource();
+        this._player.play();
     };
 
     private _setOffline = () => {
         this._broadcastState = LiveBroadcastStates.Offline;
-        if (this._overlayItem) {
+        if (
+            this._overlayItem ||
+            (this._player.config.sources.dvr && this._player.currentTime > 0)
```

Comment:
@semarche-kaltura perhaps we can add a short comment stating why we are testing currentTime > 0 
I assume that if we had reached the end / replay screen the value of currentTime is 0. Is this correct?